electrum_nmc/electrum/gui/qt/main_window.py

```diff
@@ -3367,23 +3419,115 @@ def create_buy_names_tab(self):
 
         self.buy_names_register_button = self.buy_names_ui.registerNameButton
         self.buy_names_register_button.clicked.connect(self.register_new_name)
-        self.buy_names_register_button.hide()
 
         self.buy_names_buy_button = self.buy_names_ui.buyNameButton
         self.buy_names_buy_button.clicked.connect(self.buy_name)
-        self.buy_names_buy_button.hide()
+        
+        self.gift_names_button = self.buy_names_ui.btnGiftButton
+        self.gift_names_button.clicked.connect(self.request_new_name)
+
+        self.requestnameLabel = self.buy_names_ui.requestnameLabel
+        self.ExtraAmount_Label = QLabel(_('Extra Amount:'))
+        self.extra_amount_e = BTCAmountEdit(self.get_decimal_point)
+
+        self.gift_layout = self.buy_names_ui.horizontalLayout_giftName
+        self.gift_layout.insertWidget(1, self.ExtraAmount_Label)
+        self.gift_layout.insertWidget(2, self.extra_amount_e)
+
+        self.hide_register_ui()
 
         return w
+    
+    def hide_register_ui(self):
+        self.buy_names_register_button.hide()
+        self.gift_names_button.hide()
+        self.ExtraAmount_Label.hide()
+        self.extra_amount_e.hide()
+        self.extra_amount_e.clear()
+        self.buy_names_buy_button.hide()
+        self.requestnameLabel.clear()
+        self.requestnameLabel.hide()
+
+    def show_register_ui(self):
+        self.buy_names_register_button.show()
+        self.gift_names_button.show()
+        self.ExtraAmount_Label.show()
+        self.extra_amount_e.show()
+        self.requestnameLabel.show()
+
+    def request_new_name(self):
+        try:
+            identifier_ascii = self.buy_names_new_name_lineedit.text()
+            identifier = name_from_str(identifier_ascii, Encoding.ASCII)
+            identifier_hex = name_to_str(identifier, Encoding.HEX)
+            identifier_formatted = format_name_identifier(identifier)
+
+            name_new = self.console.namespace.get('name_new')
+            addr = self.address_creation()
+
+            name_new_result = name_new(identifier=identifier_hex, name_encoding=Encoding.HEX, destination=addr, unsigned=True, commitment_only=True)
+            commitment = name_new_result["commitment"]
+
+            amount = self.extra_amount_e.get_amount() or 0
+            message = f"Pre-registration: {identifier_formatted}"
```

Comment:
Does this end up in the memo field? If so it shouldn't have the identifier in it. (Same issue as above.)

Reply:
Yes